emukit/core/discrete_parameter.py

```diff
@@ -0,0 +1,32 @@
+from typing import Iterable
+
+
+class DiscreteParameter(object):
+    """
+    A parameter that takes a discrete set of values where the order is important
```

Comment:
Is only the order important or also the spacing? Let's say we have {1, 2, 3}. Is this equivalent to {1, 2, 4}?

Reply:
Yep, spacing is important, I'll add this